src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisitionExtensionOptions.cs

```diff
@@ -38,12 +39,9 @@ internal void InvokeOnBeforeTokenAcquisitionForApp(AcquireTokenForClientParamete
         /// Invoke the BeforeTokenAcquisitionForTestUser event.
         /// </summary>
         internal void InvokeOnBeforeTokenAcquisitionForTestUser(AcquireTokenByUsernameAndPasswordConfidentialParameterBuilder builder,
-                                                           AcquireTokenOptions? acquireTokenOptions)
+                                                           AcquireTokenOptions? acquireTokenOptions, ClaimsPrincipal user)
```

Comment:
FYI, that this is a breaking change. Had we already shipped, we would have had to take a major version.

Reply:
Also, has this been tested? Surprised this wasn't caught previously.

Reply:
Agree that this is a breaking change. I noticed this needs to be added when implementing the user FIC scenario in MISE for which this extensibility is being added. I did add a unit test but the complete test will be in MISE as we cannot have an integration test for user fic in this repo. MISE and Id web are new code base for me. Sorry I missed this before.

Reply:
We can introduce as non-breaking if needed, but it's more work + soft deprecation etc.
It would be a second callback that fire.

Reply:
Breaking change is consistent across libraries, so it's not library specific. Any change in the public API (not an addition) is considered a breaking change. I'm more concerned that E2E testing has not been done to ensure there are not more changes needed to the public API. Ex. how was this not caught previously before merging?

Reply:
Discussed offline.

Reply:
@neha-bhargava What was the result of the conversation? Is this delegate only used by us?

Reply:
Ok, I see that the delegate was added recently and has not been released yet; so not a breaking change. Similar comment to Jenny's then is just making sure this was tested E2E and is all that is needed so we don't need to update it after the release.